src/lib/stores/cart.ts

```diff
@@ -0,0 +1,68 @@
+// Create a derived store for total item count
+import { derived, writable } from 'svelte/store';
+import type { Cart, Product } from '../../utils/products';
+
+const storedCart = typeof window !== 'undefined' ? localStorage.getItem('cartValue') : null;
+const initialCart: Cart[] = storedCart ? JSON.parse(storedCart) : [];
+
+const cartStore = writable<Cart[]>(initialCart);
```

Comment:
Lets use Svelte 5 features instead, Runes. In general, it might be good to have one big PR which migrates everything to new svelte5 syntax.